### ISSUE TITLE
Crash when StopCapture on iOS when swiping to see notifications

### ISSUE THREAD

**Issue by artoonie** [RenderHeads/UnityPlugin-AVProMovieCapture#125]
**Describe the bug**
Hard crash on iOS when you call StopCapture in OnApplicationFocus(false) - but only when user swipes down on notification center, not when they swap between apps.

**Your Setup (please complete the following information):**
 - Unity version: 2020.3.13f11
 - AVPro Movie Capture version: 4.6.5
 - Operating system version: iOS 14
 - Capture component used: Any (though, specifically tested CaptureFromScreen and FromTexture)
 - Capture settings (resolution, frame-rate, codec): Default settings (1080p, 30fps, h264)

**To Reproduce**
1. Script: OnApplicationPause(false), call StopCapture()
2. Open a demo scene and begin recording on iOS
3. Swipe up to go to home screen
4. Observe: no crash
5. Re-open app
6. Swipe down to view notification center
7. Observe: crash

**Logs**
```
2021-09-20 11:09:59.917656-0400 I[35563:2202297] [Plugin] 🔹 Stopping capture...
2021-09-20 11:09:59.921793-0400 I[35563:2202297] [Plugin] 🟩 Capture failed, error: The operation could not be completed
2021-09-20 11:09:59.923796-0400 I[35563:2202297] [Plugin] 🟩 Underlying error: The operation couldn’t be completed. (OSStatus error -12142.)
2021-09-20 11:09:59.924983-0400 I[35563:2202297] [Plugin] 🟩 Video capture finished.
2021-09-20 11:09:59.925021-0400 I[35563:2202297] [Plugin] 🔹 Stopped, writing file...
2021-09-20 11:09:59.940085-0400 I[35563:2202297] Uncaught exception: NSInvalidArgumentException: *** -[AVAssetWriter endSessionAtSourceTime:] invalid parameter not satisfying: CMTIME_IS_NUMERIC(endTime)
(
	0   CoreFoundation                      0x000000018a856dd4 A408A063-A023-3AAA-A674-0FF01607F818 + 1220052
	1   libobjc.A.dylib                     0x000000019f3ae7a8 objc_exception_throw + 60
	2   AVFCore                             0x0000000192309f7c 3CE02286-E97A-3C2A-9913-66A6E8EECB50 + 548732
	3   AVProMovieCapture                   0x0000000105c89af4 AVProMovieCapture + 39668
	4   AVProMovieCapture                   0x0000000105c8ecd8 Stop + 88
	5   UnityFramework                      0x000000010a3bd8cc CaptureBase_StopCapture_m1D82ABCBFF2E54B181FEF562B2F12FACA617F8C3 + 392
```

**Screenshots**

**Videos**
If applicable, add a copy of your video or the URL

**Reproducible Code**
Attach this monobehavior onto a demo scene.
```
using UnityEngine;
using RenderHeads.Media.AVProMovieCapture;

public class TestScene : MonoBehaviour
{
    void OnApplicationFocus(bool focused)                                                                                                                                                                                 
    {   
        // When app goes to background, stop recording and start it again later
        if (!focused)
        {   
            GetComponent<CaptureFromScreen>().StopCapture();
        }   
        else
        {   
            GetComponent<CaptureFromScreen>().StartCapture();
        }   
    }   
}
```

I am testing with the following demo scene:
1. A canvas with CaptureFromScreen, CaptureGUI, and the above script attached to it
2. A camera
3. An eventsystem

I can attach a demo scene as a .zip if needed.

**Comment by artoonie** [comment]
Additional info: it seems this also happens when you get a Low Power Notification, though that use case is harder to reproduce.

**Comment by bakwc** [comment]
Hi. Is there any fix or workaround?

**Comment by MorrisRH** [comment]
The problem seems to be caused by OnApplicationFocus being called multiple times in quick succession. I'm seeing the following in my test:
```
OnApplicationFocus(False) - stops the current capture
OnApplicationFocus(True) - starts a new capture
OnApplicationFocus(False) - immediately stops the new capture
```
That last OnApplicationFocus(False) is what causes the crash due to nothing actually having been captured. I'm currently working on a fix for this.

A work around would be to delay the StartCapture call in OnApplicationFocus(True) and to make sure it's not called if an OnApplicationFocus(False) is called almost immediately after.

**Comment by artoonie** [comment]
Nicely spotted. I am seeing the same, and will add a workaround until the bugfix comes in. Thanks!

**Comment by MorrisRH** [comment]
The crash will be fixed in the next release (version 4.6.6).
